docs/dcp2_system_design.rst

```diff
@@ -983,6 +1058,389 @@ running the import tool with be requested manually in
+   from the value of the ``links_id`` column in any row of the ``links`` table
+   of the TDR dataset the staging are is imported into.
+
+
+Add a project
+~~~~~~~~~~~~~
+
+1. Add the ``project`` entity as described in `Add an entity`_   
+
+2. Add all other entities belonging to the project as described in `Add an
+   entity`_ 
+
+3. Add the subgraphs belonging to the project as described in `Add a
+   subgraph`_
+
+
+Add an entity to existing subgraphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Add the entity as described in `Add an entity`_
+
+2. Update the subgraphs that reference the entity as described in `Update a
+   subgraph`_
+
+
+Add a data file to existing subgraphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Add the data file as described in `Add a data file`_
+
+2. Update the subgraphs that reference the data file as described in `Update a
+   subgraph`_
+
+
+Remove an entity
+~~~~~~~~~~~~~~~~
+
+1. Make sure the staging area property ``is_delta`` is ``true``.
+
+2. Create an empty object in the ``metadata`` directory of the staging area.
+   The ``entity_type`` and ``entity_id`` fields in the object name must be the
+   same as those in the name of the original (the entity to be deleted), the
+   ``version`` field must be strictly higher and the name must end in
+   ``.json.remove`` instead of just ``.json``.
+
+3. If the removed entity is of type ``project``, remove all subgraphs in that
+   project. Otherwise, update the subgraphs referencing the removed entity as
+   described in `Update a subgraph`_
+
+A removed entity must remain in all snapshots that are part of already
+published data releases but must be absent from any subsequently created
+snapshot in the data release currently being prepared.
+
+|nn| This means that the corresponding row cannot be removed from the TDR
+dataset, but instead has to be soft-deleted, so it is not included in
+subsequently created snapshots. This is a TDR implementation detail. |ne|
+
+
+Remove a data file
+~~~~~~~~~~~~~~~~~~
+
+1. Remove the entity describing the data file as described in `Remove an
+   entity`_. 
+
+2. Create an empty object in the ``descriptors`` directory of the staging
+   area. The name of the object must have the same ``entity_type``, ``entity_id``
+   and ``version`` as the entity from the previous step and the name must end in
+   ``.json.remove`` instead of just ``.json``.
+
+A removed data file must remain in all snapshots that are part of already
+published data releases but must be absent from any subsequently created
+snapshot in the data release currently being prepared.
+
+
+Delete a data file
+~~~~~~~~~~~~~~~~~~
+
+Follow the steps in `Remove a data file` but use ``.json.delete`` at the end
+of the descriptor object name, instead of ``.json.remove``.
+
+The importer must ensure that any copies of the data file are physically
+deleted. The metadata entity describing the deleted data file remains in
```

Comment:
Above my paygrade, but I'm wondering what happens when metadata needs to be deleted.

Reply:
It would be hard, since metadata is copied and transformed in so many ways. I'm hoping we'll never need to do that.